src/Compilers/Core/AnalyzerDriver/AnalyzerManager.AnalyzerAndOptions.cs

```diff
@@ -18,7 +18,13 @@ public AnalyzerAndOptions(DiagnosticAnalyzer analyzer, AnalyzerOptions analyzerO
                 Debug.Assert(analyzerOptions != null);
 
                 Analyzer = analyzer;
-                _analyzerOptions = analyzerOptions;
+                // _analyzerOptions is used only as a key for equality testing, and only the properties
```

Comment:
@heejaechang No, this change won't break those analyzers. AnalyzerOptions created here by John here is just used for the key of a map, not actually passed to any analyzer. They will get original instance.

@JohnHamby  which test failed? It should be unrelated to this change.